packages/actor-optimize-query-operation-filter-pushdown/test/ActorOptimizeQueryOperationFilterPushdown-test.ts

```diff
@@ -269,7 +271,77 @@ describe('ActorOptimizeQueryOperationFilterPushdown', () => {
             AF.createTermExpression(DF.variable('s')),
           ],
         ));
-        expect(actor.shouldAttemptPushDown(op, [], new Map())).toBeTruthy();
+        expect(actor.shouldAttemptPushDown(op, [], new Map(), new ActionContext())).toBeTruthy();
+      });
+
+      it('returns false if comunica supports the extensionFunction, but a source doesnt', async() => {
+        const op = AF.createFilter(
+          AF.createNop(),
+          AF.createNamedExpression(DF.namedNode('https://example.com/functions#mock'), [
+            AF.createTermExpression(DF.variable('a')),
+            AF.createTermExpression(DF.variable('b')),
+          ]),
+        );
+        const src = <any> {};
+        const context = new ActionContext().set(KeysInitQuery.extensionFunctions, {
+          'https://example.com/functions#mock': async args => args[0],
+        });
+        expect(actor.shouldAttemptPushDown(op, [ src ], new Map(), context)).toBeFalsy();
+      });
+
+      it('returns true if both comunica and all sources support the extensionFunction', async() => {
+        const op = AF.createFilter(
+          AF.createNop(),
+          AF.createNamedExpression(DF.namedNode('https://example.com/functions#mock'), [
+            AF.createTermExpression(DF.variable('a')),
+            AF.createTermExpression(DF.variable('b')),
+          ]),
+        );
+        const src: IQuerySourceWrapper = {
+          source: new QuerySourceSparql(
+            'https://example.com/src',
+            new ActionContext(),
+            <any> {},
+            'values',
+            <any> {},
+            <any> {},
+            <any> {},
+            false,
+            64,
+            10,
+            true,
+            true,
+            0,
+            undefined,
+            undefined,
+            undefined,
+            [ 'https://example.com/functions#mock' ],
+          ),
+        };
+        const shapes = new Map();
+        shapes.set(src, {
```

Comment:
Does the value object here satisfy some interface/ type? In that case you could annotate it as such: `{ ... } satisfies MyType`